### PR TITLE
mua: add Date and Message-ID headers when missing

### PR DESCRIPTION
**Pull request by ruslandoga**
👋 

This PR ensures the "rendered" message has Date and Message-ID headers set. I haven't experienced it myself, but their absence seems to be causing issues for users who send emails directly (not via a relay which usually fills in these missing headers): https://github.com/ruslandoga/bamboo_mua/issues/53

### CODE REVIEW

File: lib/swoosh/adapters/mua.ex [hunk 1]
```diff
@@ -211,8 +211,27 @@ defmodule Swoosh.Adapters.Mua do
...
+
+  defp message_id do
+    Base.hex_encode32(
+      <<
+        System.system_time(:nanosecond)::64,
+        :erlang.phash2({node(), self()}, 16_777_216)::24,
+        :erlang.unique_integer()::32
+      >>,
+      case: :lower
+    )
```
Comment (ruslandoga) [inline lib/swoosh/adapters/mua.ex:235]: I copied this from [Joken](https://github.com/joken-elixir/joken/blob/de299190741a7991dd2e66da98ec3f9d91fda737/lib/joken.ex#L190-L204) which itself is inspired by [Plug.RequestId](https://github.com/elixir-plug/plug/blob/v1.16.1/lib/plug/request_id.ex#L81-L89) 

The generated IDs that look like this:

```
2vm6j5a0ddqhnr8pdc000032
2vm6j5a0dg049r8pdc000042
2vm6j5a0dick7r8pdc000052
2vm6j5a0dililr8pdc000062
2vm6j5a0dithrr8pdc000072
2vm6j5a0dj6lhr8pdc000082
2vm6j5a0djditr8pdc000092
2vm6j5a0dk0k9r8pdc0000a2
2vm6j5a0dk8e9r8pdc0000b2
2vm6j5a0dkgntr8pdc0000c2
```

**Comment by princemaple** [comment]
Cheers, looks good

**Comment by liamwhite** [comment]
@ruslandoga the message-id format here is invalid, should be rendered in the form of \<#{id}@#{address}\>

**Comment by liamwhite** [comment]
Hi @ruslandoga, thanks for the quick fix. It causes MXRoute's signing relay to fail because it validates the Date and Message-ID.

**Pull request merged**